src/boot/efi/efi-string.c

```diff
@@ -216,6 +216,19 @@ char16_t *xstrn8_to_16(const char *str8, size_t n) {
         return str16;
 }
 
+char *startswith8(const char *s, const char *prefix) {
+        size_t l;
+
+        assert(s);
+        assert(prefix);
```

Comment:
The efi string functions accept NULL pointers everywhere, so please stick with that.

Also, this should get a test case too.